lib/b2_client/backend/httpoison.ex

```diff
@@ -107,6 +123,35 @@ defmodule B2Client.Backend.HTTPoison do
     end
   end
 
+  def download_head(b2, file_id) do
+    uri = get_download_url(b2, file_id)
+
+    case head(uri, headers(:head, b2), []) do
+      {:ok, %{status_code: 200, headers: headers}} ->
+        {_, size} =
+          Enum.find(headers, {nil, 0}, fn {key, _} ->
+            String.downcase(key) == "content-length"
+          end)
+        {_, filename} =
+          Enum.find(headers, {nil, 0}, fn {key, _} ->
+            String.downcase(key) == "x-bz-file-name"
+          end)
+
+        {:ok, %{
+            size: String.to_integer(size),
+            filename: filename
+          }
+        }
+
+      {:ok, %{status_code: code, body: original_body}} ->
+        body = Jason.decode!(original_body)
```

Comment:
IIRC a HEAD request has no body, even in error cases like 40x or 5xx?

Reply:
Yes, HEAD request has no body, docs here: https://hexdocs.pm/httpoison/HTTPoison.html#head/3

I will remove the body case in both download_head methods